cypress/integration/distroSettings/provider_section.ts

```diff
@@ -83,4 +83,94 @@ describe("provider section", () => {
       cy.validateToast("success");
     });
   });
+
+  describe("ec2 fleet", () => {
+    beforeEach(() => {
+      cy.visit("/distro/ubuntu1804-workstation/settings/provider");
+    });
+
+    it("shows and hides fields correctly", () => {
+      // Fleet options.
+      cy.getInputByLabel("Fleet Instance Type").contains("On-demand");
+      cy.dataCy("capacity-optimization").should("not.exist");
+
+      cy.selectLGOption("Fleet Instance Type", "Spot");
+      cy.dataCy("use-capacity-optimization").should("exist");
+
+      // VPC options.
+      cy.dataCy("use-vpc").should("be.checked");
+      cy.contains("Default VPC Subnet ID").should("exist");
+      cy.contains("VPC Subnet Prefix").should("exist");
+
+      cy.dataCy("use-vpc").uncheck({ force: true });
+      cy.contains("Default VPC Subnet ID").should("not.exist");
+      cy.contains("VPC Subnet Prefix").should("not.exist");
+    });
+
+    it("successfully updates ec2 fleet provider fields", () => {
+      cy.dataCy("provider-select").contains("EC2 Fleet");
+
+      // Correct section is displayed.
+      cy.dataCy("ec2-fleet-provider-settings").should("exist");
+      cy.dataCy("ec2-fleet-provider-settings")
+        .children()
+        .should("have.length", 1);
+      cy.dataCy("region-select").contains("us-east-1");
+
+      // Change field values.
+      cy.selectLGOption("Region", "us-west-1");
+      cy.getInputByLabel("SSH Key Name").clear();
+      cy.getInputByLabel("SSH Key Name").type("my ssh key");
+      cy.selectLGOption("Fleet Instance Type", "Spot");
+      cy.contains("button", "Add mount point").click();
+      cy.getInputByLabel("Device Name").type("device name");
+      cy.getInputByLabel("Size").type("200");
+      save();
+      cy.validateToast("success");
+
+      // Revert fields to original values.
+      cy.selectLGOption("Region", "us-east-1");
+      cy.getInputByLabel("SSH Key Name").clear();
+      cy.getInputByLabel("SSH Key Name").type("mci");
+      cy.selectLGOption("Fleet Instance Type", "On-demand");
+      cy.dataCy("mount-points").within(() => {
+        cy.dataCy("delete-item-button").click();
+      });
+      save();
+      cy.validateToast("success");
```

Comment:
I think I want to leave this one as-is since I would have to update multiple other test files for consistency (also I'm not sure that checking the copy is too important, it says the same thing every time)